src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/service/AddressLookupService.kt

```diff
@@ -0,0 +1,10 @@
+package uk.gov.justice.digital.hmpps.historicalprisonerapi.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.historicalprisonerapi.repository.AddressLookupRepository
+
+@Service
+class AddressLookupService(private val addressLookupRepository: AddressLookupRepository) {
+  fun findPrisonersWithAddresses(addressTerms: String): List<String> =
+    addressLookupRepository.findPrisonersWithAddresses("NEAR(($addressTerms), 5, TRUE)")
```

Comment:
think I'll need to change `addressTerms` to be parsed, but will iterate on it

Reply:
Do you think we have any SQL injection type issues here?